chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -157,81 +157,118 @@ pub struct TransactionVerifierDelta {
     token_issuance_cache: ConsumedTokenIssuanceCache,
 }
 
+/// [`TxIndexCache`] that can be enabled or disabled (using a config).
+pub struct GuardedTxIndexCache {
+    enabled: bool,
```

Comment:
Maybe `Option<TxIndexCache>`?

Reply:
It's supposed to be fixed once the node is initialized and never change unless a reindexing happens.

Reply:
If I wasn't clear, yes, it should be set on start up but never change later.

Reply:
> If I wasn't clear, yes, it should be set on start up but never change later.

It's still not quite clear to me whether it is a coding time decision, with some instances of tx verifier having the index and others lacking it, or it is something user can set when starting the node e.g. via config or command line.

Reply:
Yep. It's configurable. A transaction index is only required if you want to index transactions so that you can access all transaction history using their id. This is usually only required for block explorers, not normal users. Normal users usually have their transactions indexed by their wallets.

So, as a user, you set whether you want the transaction index to be supported when you start the node the first time, and we expect that to never change. Changing that requires to dump the current index completely, and reindex from genesis. This cannot be changed in the middle of operation or even when you start the node after shutting it down. A full reindex is required.